cmd/api/src/database/oidc.go

```diff
@@ -20,20 +20,35 @@ import (
 	"context"
 
 	"github.com/specterops/bloodhound/src/model"
+	"gorm.io/gorm"
+)
+
+const (
+	oidcProvidersTableName = "oidc_providers"
 )
 
 // OIDCProviderData defines the interface required to interact with the oidc_providers table
 type OIDCProviderData interface {
-	CreateOIDCProvider(ctx context.Context, name, issuer, clientID string) (model.OIDCProvider, error)
+	CreateOIDCProvider(ctx context.Context, name, slug, issuer, clientID string) (model.OIDCProvider, error)
 }
 
-// CreateOIDCProvider creates a new entry for an OIDC provider
-func (s *BloodhoundDB) CreateOIDCProvider(ctx context.Context, name, issuer, clientID string) (model.OIDCProvider, error) {
-	provider := model.OIDCProvider{
-		Name:     name,
+// CreateOIDCProvider creates a new entry for an OIDC provider as well as the associated SSO provider
+func (s *BloodhoundDB) CreateOIDCProvider(ctx context.Context, name, slug, issuer, clientID string) (model.OIDCProvider, error) {
+	oidcProvider := model.OIDCProvider{
```

Comment:
Consider removing the `name` property from the OIDCProvider model since it not going to be in the table

Reply:
Thought I caught all the instances of it 😭 
Good catch!

Reply:
It's removed from the model, but I still have it passed into the method since it is required for the `CreateSSOProvider` method